test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -499,7 +499,9 @@ protected ScriptService getMockScriptService() {
             InternalAggregationTestCase.assertMultiBucketConsumer(agg, shardBucketConsumer);
         }
         if (aggs.isEmpty()) {
-            return null;
+            root.preCollection();
```

Comment:
This should all the work for returning an empty aggregation. I am not 100% sure pre-pos-Collection is needed. Feedback is very welcome.

Reply:
What about just calling `aggregator.buildEmptyAggregation`? I think this one shouldn't need the pre and post stuff. Probably.

Reply:
`aggregator.buildEmptyAggregation()` seems to be working fine. thanks!